### PR TITLE
Add docker sync test to capsule

### PR DESCRIPTION
**Pull request by vsedmik**
Adding end to end test for synchronization of docker-type repositories with multiple schema versions to the capsule and pull from capsule to a content host (support for `docker login` and `docker search` are new in 6.10 capsule's Container_Gateway, opposed to Crane in 6.9).

In this PR I'm moving the `docker_host` fixture from `tests/foreman/cli/test_docker.py` to `pytest_fixtures/content_hosts.py` so that I can use it too in `tests/foreman/api/test_contentmanagement.py`.
@JacobCallahan @mshriver please ack if it's appropriate location or approach
@spusateri I ran `cli/test_docker.py::TestDockerClient::test_positive_pull_image` with this change and it passed

Also after discussion with @latran I'm moving the `TestCapsuleContentManagement` class under the Capsule-Content component and assigning it to myself.

Test result:
```
(venv39) [vsedmik@localhost robottelo]$ pytest tests/foreman/api/test_contentmanagement.py::TestCapsuleContentManagement::test_positive_sync_container_repo_end_to_end
=========================================================================================================== test session starts ===========================================================================================================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/vsedmik/PycharmProjects/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, xdist-2.4.0, reportportal-5.0.8, ibutsu-1.16, mock-3.6.1
collected 1 item                                                                                                                                                                                                                          

tests/foreman/api/test_contentmanagement.py .                                                                                                                                                                                       [100%]
...
=============================================================================================== 1 passed, 3 warnings in 3332.51s (0:55:32) ================================================================================================
```

### CODE REVIEW

**Comment by vsedmik** [comment]
In the second commit I've extended the test by podman client, so that we test both, `docker` and `podman` on the same `container_contenthost`.
Tried to run @spusateri 's tests to ensure I don't break them by having both clients on the host. They succeeded:
```
$ pytest tests/foreman/cli/test_docker.py::TestDockerClient::test_positive_pull_image tests/foreman/cli/test_docker.py::TestDockerClient::test_positive_container_admin_end_to_end_search tests/foreman/cli/test_docker.py::TestDockerClient::test_positive_container_admin_end_to_end_pull
=========================================================================================================== test session starts ===========================================================================================================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/vsedmik/PycharmProjects/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, xdist-2.4.0, reportportal-5.0.8, ibutsu-1.16, mock-3.6.1
collected 3 items                                                                                                                                                                                                                         

tests/foreman/cli/test_docker.py ...                                                                                                                                                                                                [100%]

=============================================================================================== 3 passed, 3 warnings in 1922.04s (0:32:02) ================================================================================================
```

File: pytest_fixtures/content_hosts.py [hunk 1]
```diff
@@ -95,3 +96,20 @@ def registered_hosts(organization_ak_setup, content_hosts, default_sat):
         vm.register_contenthost(org.label, ak.name)
         assert vm.subscribed
     return content_hosts
+
+
+@pytest.fixture()
```
Comment (JacobCallahan) [inline pytest_fixtures/content_hosts.py:101]:
```suggestion
@pytest.fixture
```

File: tests/foreman/api/test_contentmanagement.py [hunk 2]
```diff
@@ -995,3 +999,110 @@ def test_positive_sync_kickstart_repo(
...
+        :expectedresults:
+            1. The container repo is successfully synced to the Capsule.
+            2. The image is successfully pulled from Capsule to a host.
+
+        :CaseLevel: Integration
+        """
+        upstream_names = [
+            'quay/busybox',  # v1
+            'foreman/busybox-test',  # v2
+            'foreman/foreman',  # v1+v2
```
Comment (JacobCallahan) [inline tests/foreman/api/test_contentmanagement.py:1029]: what do the v1 and v2 mean in these comments?

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Small things

File: tests/foreman/api/test_contentmanagement.py (hunk 2 above)
Comment (vsedmik) [reply tests/foreman/api/test_contentmanagement.py]: It refers to the schema versions available in the image.

**Comment by vsedmik** [comment]
rebased

File: tests/foreman/cli/test_docker.py [hunk 3]
```diff
@@ -1495,7 +1484,7 @@ def test_positive_upload_image(self, module_org, default_sat):
             compute_resource = make_compute_resource({
                 'organization-ids': [module_org.id],
                 'provider': 'Docker',
-                'url': f'http://{docker_host.ip_addr}:2375',
+                'url': f'http://{docker_contenthost.ip_addr}:2375',
```
Comment (pondrejk) [inline tests/foreman/cli/test_docker.py:1487]: should this be container_contenthost?

**Review by pondrejk — COMMENTED** [review]
ack pending one query

File: tests/foreman/cli/test_docker.py (hunk 3 above)
Comment (vsedmik) [reply tests/foreman/cli/test_docker.py:1487]: Oh, how could I have missed that? Good catch!

Comment (vsedmik) [reply tests/foreman/cli/test_docker.py]: Probably because it was inside a comment and refactor does not edit comments...
Fixed.

**Pull request merged**